packages/myst-cli/src/project/toTOC.ts

```diff
@@ -38,50 +19,119 @@ function chaptersFromPages(pages: (LocalProjectFolder | LocalProjectPage)[], pat
- */
-export function tocFromProject(project: PartialLocalProject, path = '.') {
-  const toc: TOC = {
-    format: 'jb-book',
-    root: getRelativeDocumentLink(project.file, path),
-    chapters: chaptersFromPages(project.pages, path),
-  };
-  return toc;
+export function tocFromProject(
+  project: Omit<LocalProject, 'bibliography'>,
+  dir: string = '.',
+): any {
+  return [
+    { file: getRelativeDocumentLink(project.file, dir) },
+    ...tocFromPages(project.pages, dir),
+  ];
 }
 
-export function writeTOCFromProject(project: PartialLocalProject, path: string) {
-  const filename = join(path, '_toc.yml');
-  const content = `${GENERATED_TOC_HEADER}${yaml.dump(tocFromProject(project, path))}`;
-  fs.writeFileSync(filename, content);
+const DEFAULT_INDENT = 2;
+
+export async function writeTOCToConfigFile(
+  project: Omit<LocalProject, 'bibliography'>,
+  srcPath: string,
+  dstPath: string,
+) {
+  // Read the existing configuration file
+  let lines: string[];
+  if (fs.existsSync(srcPath)) {
+    lines = fs.readFileSync(srcPath, 'utf8').split(/\r\n|\r|\n/);
+  } else {
+    lines = [];
+  }
+
+  // Identify the first non-empty line's indentation
+  const indent = lines.map(yamlLineIndent).find((item) => item !== undefined) ?? 0;
+
+  // Find `project` section
+  let projectInfo = findYAMLSection('project', indent, lines);
+
+  // Or, create an empty project section
+  if (!projectInfo) {
+    const prefix = ' '.repeat(indent);
+    lines.push(`${prefix}project:`);
+    projectInfo = { start: lines.length, stop: lines.length, indent: indent + DEFAULT_INDENT };
+  }
+
+  // Identify the child indentation of the project section
+  const projectIndent = projectInfo.indent ?? indent + DEFAULT_INDENT;
+
+  // Find `toc` section
+  let tocInfo = findYAMLSection('toc', projectIndent, lines, projectInfo.start);
+
+  // Or create an empty toc section
+  if (!tocInfo) {
+    const prefix = ' '.repeat(projectIndent);
+    lines.splice(projectInfo.stop, 0, `${prefix}toc:`);
+
+    // Append project info
+    tocInfo = {
+      start: projectInfo.stop + 1,
+      stop: projectInfo.stop + 1,
+      indent: projectIndent + DEFAULT_INDENT,
+    };
+  }
+
+  // Determine indentation of section
+  const tocIndent = tocInfo.indent ?? projectIndent + DEFAULT_INDENT;
+  const tocPrefix = ' '.repeat(tocIndent);
+
+  // Build the new `toc` section
+  const { dir } = parse(srcPath);
+  const toc = tocFromProject(project, dir);
+  const tocLines = [
+    `${tocPrefix}# Auto-generated by \`myst init --write-toc\``,
+    ...yaml
+      .dump(toc)
+      .split(/\r\n|\r|\n/)
+      .map((line) => `${tocPrefix}${line}`),
+  ];
+  // Splice these `toc` lines into the proper location
+  lines.splice(tocInfo.start, tocInfo.stop - tocInfo.start, ...tocLines);
+
+  // Write the new configuration
+  const newConfigContent = lines.join('\n');
+
+  // Ensure the hand-rolled YAML is valid
+  try {
+    yaml.load(newConfigContent);
+  } catch (err) {
+    throw new Error(
```

Comment:
Can we log the expected yaml to add here in a comment, so the logs are helpful, for example:

```
We could not automatically add the TOC to your `myst.yml`, it is here:

...
```